GVFS/GVFS.Installer.Mac/scripts/postinstall

```diff
@@ -1,24 +1,39 @@
 #!/bin/bash
 
+LOGFILENAME=`/bin/date +$HOME/Library/Application\ Support/GVFS/GVFS.Upgrade/UpgraderLogs/gvfs_installer_postinstall_%Y%m%d_%H%M%S.log`
```

Comment:
Does this work fine if we don't already have the `GVFS/GVFS.Upgrade` folder in /Library/Application Support?
What's the permissions on this folder if it's created by the postinstall script? Will it be owned by root and therefore un-writable by other parts of the installer/upgrader process?

Reply:
good question!

Can we control the location of the logs from the command line? When running upgrade, we might want the logs to go into the "upgrader" directory. But if we are just running the installer directly, we might not want the logs there... On the Windows side of things, the upgrader app passes in the log directory. Can we do something similar here?

This also hardcodes the location of the installer logs here to match what is in code - so if we ever update the log directory, it we will need to remember that there are 2 locations to update.

Reply:
>> Does this work fine if we don't already have the GVFS/GVFS.Upgrade folder in /Library/Application Support?

No it doesn't - fresh installs would have failed (in-fact [one of our pipeline builds](https://dev.azure.com/mseng/AzureDevOps/_build/results?buildId=10232664&view=results) were failing for this reason). Added [new commit](https://github.com/microsoft/VFSForGit/pull/1466/commits/12f29c829b716713d91b64d788ba4f86056799cd) that creates the directory if it does not exist.

>> What's the permissions on this folder if it's created by the postinstall script? Will it be owned by root and therefore un-writable by other parts of the installer/upgrader process?

[This commit](https://github.com/microsoft/VFSForGit/pull/1466/commits/12f29c829b716713d91b64d788ba4f86056799cd) creates the directory using `sudo -u $USER` where `USER` is an installer variable that refers to the active logged-in user (and not `root` user) - the directory gets created with the ownership and permissions of  `USER`, even when running as `root` during `sudo gvfs upgrade --confirm`.

This is how it looks on my Mac
```
$ ls -Rl ~/Library/Application\ Support/GVFS/
total 0
drwxr-xr-x  4 Ameen  staff  128 Aug 21 10:39 GVFS.Upgrade

/Users/Ameen/Library/Application Support/GVFS//GVFS.Upgrade:
total 0
drwxr-xr-x  4 Ameen  staff  128 Aug 21 10:40 UpgraderLogs
```
>>Can we control the location of the logs from the command line?

Installer does not seem to provide an option to pass custom args from command line to installer scripts.

>> This also hardcodes the location of the installer logs here to match what is in code - so if we ever update the log directory, it we will need to remember that there are 2 locations to update.

Agree.

Reply:
@jrbriggs - thoughts?

Reply:
Where would stdout go if we're not teeing to the log file? Is there an opportunity to play closer with the OS and not need to take responsibility for creating these files?

If not, this seems like the best compromise. Thanks!

Reply:
My question was more around the fact that it seemed odd that the installer would write its logs to the "upgrade" folder, even if the installation was happening outside of an `gvfs upgrade` workflow. It now creates implicit dependencies / assumptions between the `gvfs upgrade` code and the installer package.

It is a convenient way when the installer is being run as part of the upgrade, so maybe it is a good compromise.

Reply:
> It is a convenient way when the installer is being run as part of the upgrade, so maybe it is a good compromise.

Would it be possible and/or make sense to write the installer logs to `GVFS.Installer` folder (regardless of whether or not the installer is running as part of an upgrade)?  I'm not sure if that would be less confusing?  Thoughts?

Reply:
>> Where would stdout go if we're not teeing to the log file? Is there an opportunity to play closer with the OS and not need to take responsibility for creating these files?

This is possible if we capture `/var/log/install.log`. It holds error logs from `vfsforgit`'s installer scripts and also failure info from `git`, `gcm-core` installers. Compared to custom installer logs from installer scripts, this is non-lossy. It gets created only during  failures in `gvfs upgrade` (and not during manual installs using GUI.) 

Added [commit](https://github.com/microsoft/VFSForGit/pull/1466/commits/b2d69ec1c23429bec609756a9acc30c5529cd77e).